File: src/1419-Minimum-Number-of-Frogs-Croaking/1419.java
class Solution {
    public int minNumberOfFrogs(String cr) {
        int c = 0, r = 0, a = 0, k = 0, o = 0, frogs = 0, res = 0;
		for(char i : cr.toCharArray()) {
			if (i=='c') {c++; frogs++;}
			else if (i=='r') r++; 
			else if (i=='o') o++; 
			else if (i=='a') a++; 
			else {k++; frogs--;}
			res = Math.max(res, frogs);
			if (c < r || r < o || o < a || a < k) return -1;    
		}
		if (c == r && c == o && c == a && c == k) return res;
		return -1;
    }
}

File: src/1419-Minimum-Number-of-Frogs-Croaking/1419.go
func minNumberOfFrogs(cr string) int {
    res, frogs := 0, 0
    st := make([]int, 5)
    for _, i := range []byte(cr) {
        if i == 'c' {
            st[0]++
            frogs++
        } else if i == 'r' {
            st[1]++
        } else if i == 'o' {
            st[2]++
        } else if i == 'a' {
            st[3]++
        } else {
            st[4]++
            frogs--
        }
        res = max(res, frogs)
        if st[0] < st[1] || st[1] < st[2] || st[2] < st[3] || st[3] < st[4] {
            return -1
        }
    }
    
    if st[0] == st[1] && st[1] == st[2] && st[2] == st[3] && st[3] == st[4] {
        return res
    }
    return -1
}

func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}